subsys/bluetooth/mesh/pb_gatt_srv.c

```diff
@@ -115,14 +115,12 @@ static void gatt_disconnected(struct bt_conn *conn, uint8_t reason)
 
 	bt_conn_get_info(conn, &info);
 	if (info.role != BT_CONN_ROLE_PERIPHERAL || !service_registered ||
-	    info.id != BT_ID_DEFAULT) {
+	    info.id != BT_ID_DEFAULT || !cli || cli->conn != conn) {
```

Comment:
Actually, for pb_gatt there should always be exactly one connection. The provisioning process can work only with one connected client at a time. But may be this is another issue that we can fix separately.

Reply:
Yeah, this seems reasonable. But this specific change fixes an issue where an application uses the default identity to advertise connectable advertisements. Then once remote device connects or disconnects, this callback is triggered.